test/integration/errors-oidc.js

```diff
@@ -0,0 +1,112 @@
+const supertest = require('supertest')
+const ldnode = require('../../index')
+const path = require('path')
+const fs = require('fs-extra')
+const expect = require('chai').expect
+
+describe('OIDC error handling', function () {
+  process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0'
+
+  const serverUri = 'https://localhost:3457'
+  var ldpHttpsServer
+  const rootPath = path.join(__dirname, '../resources/accounts/errortests')
+  const dbPath = path.join(__dirname, '../resources/accounts/db')
+
+  const ldp = ldnode.createServer({
+    root: rootPath,
+    sslKey: path.join(__dirname, '../keys/key.pem'),
+    sslCert: path.join(__dirname, '../keys/cert.pem'),
+    auth: 'oidc',
+    webid: true,
+    idp: false,
+    strictOrigin: true,
+    dbPath,
+    serverUri
+  })
+
+  before(function (done) {
+    ldpHttpsServer = ldp.listen(3457, done)
+  })
+
+  after(function () {
+    if (ldpHttpsServer) ldpHttpsServer.close()
+    fs.removeSync(path.join(rootPath, 'index.html'))
+    fs.removeSync(path.join(rootPath, 'index.html.acl'))
+  })
+
+  const server = supertest(serverUri)
+
+  describe('Unauthenticated requests to protected resources', () => {
+    describe('accepting json only', () => {
+      it('should return 401 Unauthorized with www-auth header', done => {
```

Comment:
True - this version of `supertest` doesn't support promises I think, hence the explicit done(). I was planning on upgrading to the latest soon anyway, though.